libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.ts

```diff
@@ -0,0 +1,182 @@
+  ).toString();
+
+export const calculateRemainder = (options: {
+  actors: ActorsByActorType;
+  creditsDocumentUnitPrice: NonZeroPositive;
+  massIdTotalValue: BigNumber;
+}): Remainder => {
+  const { actors, creditsDocumentUnitPrice, massIdTotalValue } = options;
+
+  let participantsAmount = new BigNumber(0);
+  let participantsPercentage = new BigNumber(0);
+
+  for (const actor of actors.values()) {
+    participantsAmount = participantsAmount.plus(actor.amount);
+    participantsPercentage = participantsPercentage.plus(actor.percentage);
+  }
+  const rawAmount = massIdTotalValue
+    .multipliedBy(creditsDocumentUnitPrice)
+    .minus(participantsAmount);
+  const rawPercentage = new BigNumber(100).minus(participantsPercentage);
+
+  return {
+    amount: formatDecimalPlaces(BigNumber.maximum(rawAmount, 0)),
+    percentage: formatDecimalPlaces(BigNumber.maximum(rawPercentage, 0)),
+  };
+};
+
+export const calculateAmount = (options: {
+  creditsDocumentUnitPrice: NonZeroPositive;
+  massIdCertificateValue: BigNumber;
+  massIdPercentage: string;
+  participantAmount: string | undefined;
+}): NonEmptyString => {
+  const amount = formatDecimalPlaces(
+    formatPercentage(new BigNumber(options.massIdPercentage))
+      .multipliedBy(options.massIdCertificateValue)
+      .multipliedBy(new BigNumber(options.creditsDocumentUnitPrice)),
+  );
+
+  return amount.plus(options.participantAmount ?? 0).toString();
+};
+
+export const addParticipantRemainder = (options: {
+  actors: ActorsByActorType;
+  remainder: Remainder;
+}): void => {
+  const { actors, remainder } = options;
+
+  for (const [key, actor] of actors) {
+    if (actor.actorType === RewardsDistributionActorType.NETWORK) {
+      actors.set(key, {
+        ...actor,
+        amount: remainder.amount.plus(actor.amount).toString(),
+        percentage: remainder.percentage.plus(actor.percentage).toString(),
+      });
+
+      break;
+    }
+  }
+};
+
+const calculateCertificateTotalValue = (certificateValues: BigNumber[]) =>
+  sumBigNumbers(certificateValues);
+
+export const aggregateMassIdCertificatesRewards = (
+  creditsDocumentUnitPrice: NonZeroPositive,
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[],
+): AggregateMassIdRewards => {
+  const actors: ActorsByActorType = new Map();
+
+  const certificateTotalValue = calculateCertificateTotalValue(
+    resultContentsWithMassIdCertificateValue.map(
+      ({ massIdCertificateValue }) => massIdCertificateValue,
+    ),
+  );
+
+  const creditTotal = formatDecimalPlaces(
+    certificateTotalValue.multipliedBy(creditsDocumentUnitPrice),
+  );
+
+  for (const {
+    massIdCertificateValue,
+    resultContent: { massIdRewards },
+  } of resultContentsWithMassIdCertificateValue) {
+    for (const massIdReward of massIdRewards) {
+      const { actorType, massIdPercentage, participant } = massIdReward;
+      const actor = actors.get(`${actorType}-${participant.id}`);
```

Comment:
I think it is better to have a helper function `getAggregateParticipantKey({ actorType, participantId })`, and use it here and in the test,